### ISSUE TITLE
DSPACE submission via SWORD

### ISSUE THREAD

**Issue by greatislander** [pressbooks/excalibur#1]
### Prerequisites

* [x] Can you reproduce the behavior you wish to address with [all plugins except Pressbooks and this plugin deactivated](http://codex.wordpress.org/Multisite_Network_Administration#Plugins)?
* [x] Are you running the [latest version of Pressbooks](http://github.com/pressbooks/pressbooks/releases/latest/), the latest version of [this plugin](http://github.com/pressbooks/excalibur/releases/latest/) and the [latest version of WordPress](http://codex.wordpress.org/Upgrading_WordPress)?
* [x] Are you experiencing the issue while using this book theme?
* [x] Did you [perform a cursory search](https://github.com/issues?q=+is%3Aissue+repo%3Apressbooks%2Fexcalibur) to see if your bug or enhancement is already listed?

### Description

DSPACE submission via SWORD. Detailed specifications to follow; should adopt [SELF-e](http://self-e.libraryjournal.com) submission UX as a basis.

### Steps to Reproduce

1. User (with admin credentials for current book) navigates to **Publish &rarr; Submit to Repository** 
2. User fills in required information (e.g. authentication), clicks **Submit**

**Expected behavior:** Profit!

### System Information

Pressbooks: 4.0.0-RC1
WordPress: 4.8
PHP: 7.1

**Comment by connerbw** [comment]
Forked php-sword-app to be compatible with Composer [1]. Also, PSR2. Also, `composer test` works now. Also, there were bugs in the original library -- loading the files in PHPStorm highlights them (wrong params, wrong number of params, missing properties,...)

[1] https://github.com/pressbooks/swordappv2-php-library

Hopefully, [swordapp](https://github.com/swordapp) will consider merging the changes. I will put in a formal request after I am done integrating with our plugin.

**Comment by connerbw** [comment]
Working prototype. 🥇  

As I type, if you know your Deposit URL then a book can be submitted to DSpace over SWORD V1.

I abandoned trying to get SWORD V2 working. I spent many hours on this and simply couldn't get it working. We should delete `pressbooks/swordappv2-php-library` from GitHub and Packagist. 

TODO: 
 
  + Two page interface
    + Page 1: Call serviceDocument and query for Deposit URLs
    + Page 2: Same as now but  add Deposit URLs to a list the user can select from.
 + Write unit tests
 + Create test cases that demonstrate where SWORD V2 fails and submit to Dspace's issue tracke.
 + Investigate what other meta data can be submitted over SWORD V1.
